Issue26

alessandroRenzi opened the pull request:
Close #26

alessandroRenzi committed:
Added mongo2.14.3 dependency in pom.xml

alessandroRenzi committed:
Added fongo1.6.5 in pom.xml

alessandroRenzi committed:
Created DbMongoTest class, to Test DbMongo Class

alessandroRenzi committed:
Created DbMongo class, to solve TestDbMongo compile error

alessandroRenzi committed:
Created setUp method in DbMongoTest 

added constructor in DbMongo class.

alessandroRenzi committed:
Created testGetAllStudentEmpty

changed getAllStudent method in DbMongo

alessandroRenzi committed:
added testGetAllStudentsNotEmpty, all test working.

modified method getAllStudentsList in DbMongo 
modified constructor in DbMongo to get the Collection students
Added constructor in class Student

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/DbMongoTest.java:
```diff
@@ -0,0 +1,48 @@
+public class DbMongoTest {
+	private DbMongo database;
+	private DBCollection students;
+	
```
remove this additional empty line

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/DbMongoTest.java:
```diff
@@ -0,0 +1,48 @@
+		document.put("id", "1");
+		document.put("name", "first");
+		students.insert(document);
+		assertEquals(1, database.getAllStudentsList().size());
```
Besides that, it would be important to also test that the Student returned in the list has the expected values for id and name

LorenzoBettini commented on com.examples.exercise/src/main/java/com/examples/exercise/Student.java:
```diff
@@ -4,6 +4,16 @@
 	private String id;
 	private String name;
 
+	
```
Remove this additional empty line

alessandroRenzi committed:
Deleted empty lines and improved testGetAllStudentNotEmpty 

modified method test, now it's checking that if Student returned in the
list is reliable

LorenzoBettini commented:
Another PR has been merged meanwhile, which added another method to the interface Database, thus during the merging, your Database implementation does not compile anymore. You should merge master into your branch and add an empty implementation for the new method in the interface, and then push your branch again.

alessandroRenzi committed:
Merge branch 'Issue24' into issue26

alessandroRenzi committed:
Revert "changed order instruction on travis.yml"

This reverts commit f8c925db6da182ed22bdddb5d900376664b150ad.

alessandroRenzi committed:
Merge branch 'master' into issue26, and added requested method

added method in Database interface and unimplemented method in DbMongo
class

Conflicts:
	.travis.yml

LorenzoBettini commented:
Github says that it cannot merge since there are conflicts in Database.java. Please merge master into your branch and make sure you solve all the conflicts.

alessandroRenzi commented:
I have notcied that, and i'm sorry for that.

I will try to explain what i did:
First i have merged master into branch issue26 and then i found a conflict between travis.xml files, i solved that and at the same time i added the method requested in Database interface, and then committed and pulled on.
Now, if i try to merge, egit says me everything is up to date and there are no conflict.
![istantanea_2016-11-25_12-21-37](https://cloud.githubusercontent.com/assets/16014665/20623816/c47ecc0a-b309-11e6-88da-294da95f80dd.png).

LorenzoBettini commented:
from the screenshot it looks like your master is at the same level as your branch and this is wrong... you first need to fetch the master from the remote repository and merge the remote master into your branch.

alessandroRenzi commented:
when i try to fetch from master branch, eclipse says me everything is up to date, so i checked with git in terminal, and in fact, it found a diverged branch.
Git says me i have to use pull to merge it.
Is that ok?
![istantanea_2016-11-25_13-05-51](https://cloud.githubusercontent.com/assets/16014665/20624967/545f71a2-b310-11e6-9cdb-102cfeeac353.png)

LorenzoBettini commented:
You need to fetch from origin, and in your git repo you should see that origin/master is different from your current master

alessandroRenzi committed:
Merge branch 'master' into issue26


Conflicts:
	.Database.java

alessandroRenzi commented:
Thank you for helping me.
